src/ol_infrastructure/components/aws/eks.py

```diff
@@ -19,13 +19,22 @@ class OLEKSGatewayRouteConfig(BaseModel):
     backend_service_name: Optional[str]
     backend_service_namespace: Optional[str]
     backend_service_port: Optional[int]
-    certificate_secret_name: Optional[str]
-    certificate_secret_namespace: Optional[str] = ""
-    hostname: str
+    filters: Optional[list[dict[str, Any]]] = []
+    matches: Optional[list[dict[str, Any]]] = None
+    listener_name: str
+    hostnames: list[str]
+    name: str
+    port: int
+
+
+class OLEKSGatewayListenerConfig(BaseModel):
     name: str
+    hostname: str
     port: int
-    protocol: str
+    protocol: Literal["HTTPS", "HTTP"] = "HTTPS"
     tls_mode: Literal["Passthrough", "Terminate"] = "Terminate"
+    certificate_secret_name: Optional[str]
```

Comment:
The comment above to the effect that cert-manager will generate this was super helpful. Could we maybe name this field in a way that signposts that?

generated_certificate_secret_name comes to mind off the top but I'm sure folks could come up with something better :)

Reply:
Yah I was actually thinking about that too and forgot to come back to it. The (little) complication is that you don't HAVE to use cert-manager. Let me see how I can clarify it.